### DISCUSSION TITLE
Supporting other runtime platforms

### DISCUSSION THREAD

**Discussion by vngantk** [flix/flix#5200] [General]
Any plan to support other runtime platforms such as JavaScript, WASM, and LLVM?

**Comment by magnus-madsen** [comment]
We are interested in WASM. 

We are waiting for them to land tail calls and a GC.

**Comment by bikallem** [comment]
Wasm 3.0 supports both tail calls and a GC plus more. https://webassembly.org/news/2025-09-17-wasm-3.0/#:~:text=Wasm%20tail%20calls%20are%20fully,(by%20reference%20or%20table).

**Reply by magnus-madsen** [reply]
We are aware and looking into it.

**Reply by evelant** [reply]
Exciting, looks like these features are finally available by default in all major browsers and runtimes. I discovered Flix awhile back and it looks really interesting but I've held off on trying it because it's JVM only and I almost exclusively work with isomorphic code which automatically excludes all JVM stuff. I look forward to trying Flix if/when there's a wasm target!
